ffi/src/lib.rs

```diff
@@ -389,12 +394,14 @@ unsafe fn unwrap_and_parse_path_as_url(path: KernelStringSlice) -> DeltaResult<U
 }
 
 // A client builder allows setting options before building an actual client
+#[cfg(any(feature = "default-engine", feature = "sync-engine"))]
```

Comment:
The semantics will be a bit odd if you only enable `sync-engine` at the moment. You'll be able to see the types for this, but not actually get one (because `get_engine_builder` is only `#[cfg(feature = "default-engine")]`)

It's probably fine for now, but maybe make an issue to follow-up and have the engine builder be able to be correctly used to build either a `default` or `sync` engine depending on options and/or enabled flags.

Reply:
Is it enough to just make `get_engine_builder` visible under `sync-engine` as well? Or something more complicated that merits a separate follow-on effort?

Reply:
Not enough no, because we have no way of indicating that we want a sync engine. We'd have to probably make an enum with the types, and  then either have that as an arg to `get_engine_builder`, or we could have `get_default_builder` or `get_sync_builder`.

But right now the actual `build` method is hard-coded to use the default client

Reply:
Filed as https://github.com/delta-incubator/delta-kernel-rs/issues/199, thanks!